File: Day74/Part2/platform_swap.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
    int n;
    if (cin >> n) {
        if (n == 0) {
            cout << "1" << endl;
            return 0;
        }
        vector<int> h(n);
        for (int i = 0; i < n; ++i) {
            cin >> h[i];
        }

        sort(h.begin(), h.end());

        bool possible = true;
        for (int i = 0; i < n - 1; ++i) {
            if (abs(h[i] - h[i+1]) > 1) {
                possible = false;
                break;
            }
        }

        cout << (possible ? "1" : "0") << endl;
    }
    return 0;
}

File: Day74/Part2/platform_swap.py
def solve():
    try:
        line = input().split()
        if not line:
            return
        
        if len(line) == 1:
            n = int(line[0])
            line2 = input().split()
            h = [int(x) for x in line2]
        else:
            h = [int(x) for x in line]
            n = len(h)
            
        if n == 0:
            print("1")
            return
            
        h.sort()
        
        possible = True
        for i in range(n - 1):
            if abs(h[i] - h[i+1]) > 1:
                possible = False
                break
                
        print("1" if possible else "0")
            
    except EOFError:
        pass

if __name__ == "__main__":
    solve()
